qutip/solver/multitrajresult.py

```diff
@@ -804,6 +846,11 @@ def _initialize_sum_states(self, example_trajectory):
         self.sum_states = [
             qzero_like(_to_dm(state)) for state in example_trajectory.states]
 
+    def _initialize_sum_finalstate(self, example_trajectory):
+        self.sum_final_state = qzero_like(
+            _to_dm(example_trajectory.final_state)
+        )
```

Comment:
Thank you for sorting out this final state mess. I would suggest changing also lines 833-834 into
```
        if example_trajectory.final_state and store_final_state:
            self._initialize_sum_finalstate(example_trajectory)
```